tests/rados/test_noscrub_nodeepscrub_check.py

```diff
@@ -0,0 +1,218 @@
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_object = MonConfigMethods(rados_obj=rados_object)
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    scrub_obj = RadosScrubber(node=cephadm)
+
+    # Start with a cluster that has nodeep-scrub and noscrub flags set
+    scrub_obj.set_osd_flags("set", "noscrub")
+    scrub_obj.set_osd_flags("set", "nodeep-scrub")
+    if not wait_for_clean_pg_sets(rados_object, timeout=300):
+        log.error("Cluster is not in active+clean state")
+        raise Exception(
+            "Cluster is not in actve+clean state.Not executing the further tests"
+        )
+    init_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'")
+    init_time = init_time.strip()
+    # Created pool and pushed the data.
+    pool_name = config["pool_name"]
+    try:
+        method_should_succeed(rados_object.create_pool, **config)
+        log.info(f"The {pool_name} pool is created")
+        rados_object.bench_write(pool_name=pool_name, max_objs=5000)
+        log.info(f"The data is pushed into the {pool_name} pool")
+        pg_id_list = rados_object.get_pgid(pool_name=pool_name)
+        log.info(f"The pg list of the pool {pool_name} is-- {pg_id_list}")
+
+        # Modified the scrub parameters
+        mon_object.set_config(section="osd", name="osd_scrub_min_interval", value=60)
+        mon_object.set_config(section="osd", name="osd_deep_scrub_interval", value=420)
+        mon_object.set_config(section="osd", name="debug_osd", value="20/20")
+
+        # Check for log
+        exec_end_time = datetime.datetime.now() + datetime.timedelta(seconds=20)
+        while exec_end_time > datetime.datetime.now():
+            end_time, _ = installer.exec_command(
+                cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
+            )
+            end_time = end_time.strip()
+            for pg_id in pg_id_list:
+                log_lines = [
+                    rf"{pg_id}\s*(?!deep-)scrub starts",
+                    rf"{pg_id}\s*deep-scrub starts",
+                ]
+                if verify_scrub_log(
+                    rados_object, init_time, end_time, pg_id, log_lines
+                ):
+                    log.error(
+                        f"The scrub started on the {pg_id} after setting the no_scrub and no_deepscrub flag"
+                    )
+                    return 1
+            time.sleep(30)
+
+        log.info(
+            "Verify that the deep-scrub starts after unsetting the nodeep-scrub flag"
+        )
+        init_time, _ = installer.exec_command(
+            cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
+        )
+        init_time = init_time.strip()
+        scrub_obj.set_osd_flags("unset", "nodeep-scrub")
+        log.info("The nodeep-scrub flag is unset")
+        time.sleep(20)
+        deep_scrub_falg = False
+        exec_end_time = datetime.datetime.now() + datetime.timedelta(seconds=90)
```

Comment:
is this time enough for the scrubbing to start and finish on the pool?

Reply:
Yes, time is enough.